src/libcprover-cpp/api.cpp

```diff
@@ -137,6 +138,28 @@ std::unique_ptr<verification_resultt> api_sessiont::verify_model() const
 {
   PRECONDITION(implementation->model);
 
+  bool empty_result = preprocess_model();
+  if(empty_result)
+  {
+    return {};
+  }
+
+  return run_verifier();
+}
+
+void api_sessiont::read_goto_binary(std::string &file) const
+{
+  implementation->model = util_make_unique<goto_modelt>(
+    ::read_goto_binary(file, *implementation->message_handler).value());
```

Comment:
:no_entry_sign: `read_goto_binary` may return an empty optional. In that case this implementation will call `.value()` on it to get the non-existent value; which isn't exactly going to fail gracefully.

Reply:
Done (error handling added in the first commit)